src/dune_engine/vcs.ml

```diff
@@ -142,5 +151,5 @@ let files =
 
               let to_dyn = Dyn.Encoder.list Path.to_dyn
             end ))
-       ~git:(f [ "ls-tree"; "-r"; "--name-only"; "HEAD" ])
+       ~git:(f_zero [ "ls-tree"; "-z"; "-r"; "--name-only"; "HEAD" ])
        ~hg:(f [ "files" ])
```

Comment:
`hg` should be given a similar treatment with `hg files -0`.

Reply:
Done.